### PR TITLE
[ACS-4461] Added sorting in Category list for control centre

### PR DESCRIPTION
**Pull request by swapnil-verma-gl**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [x] Other... Please describe:
Category lists are now being sorted

**What is the current behaviour?** (You can also link to an open issue here)
Category list on control centre were not sorted alphabetically

**What is the new behaviour?**
Category list on control centre were not sorted alphabetically

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/content-services/src/lib/category/services/category-tree-datasource.service.ts [hunk 1]
```diff
@@ -54,6 +54,7 @@ export class CategoryTreeDatasourceService extends TreeService<CategoryNode>  {
                 };
                 nodesList.push(loadMoreNode);
             }
+            nodesList = nodesList.sort((category1, category2) => category1.nodeName.localeCompare(category2.nodeName));
```
Comment (DenysVuika) [inline lib/content-services/src/lib/category/services/category-tree-datasource.service.ts:57]: should this be sorted before you push "loadMoreNode" to the end?

File: lib/content-services/src/lib/category/services/category-tree-datasource.service.ts [hunk 2]
```diff
@@ -76,7 +77,10 @@ export class CategoryTreeDatasourceService extends TreeService<CategoryNode>  {
                 );
             }),
                 toArray(),
-                map(res =>  ({entries: res, pagination: pagingResult.list.pagination})));
+                map(res =>  {
+                    const sortedResults = res.sort((category1, category2) => category1.nodeName.localeCompare(category2.nodeName));
```
Comment (AleksanderSklorz) [inline lib/content-services/src/lib/category/services/category-tree-datasource.service.ts:81]: I don't think it is proper way to do that. We should do sorting on backend and just proper request to backend on frontend. In other case we just sort single 'page'. We have raised backend jira already for that so this should be done after that backend jira is finished: 
https://alfresco.atlassian.net/browse/ACS-4664

**Review by AleksanderSklorz — CHANGES_REQUESTED** [review]
Marking as 'request changes' as this change should be done on backend and we should to just integrate with backend by doing proper request.

File: lib/content-services/src/lib/category/services/category-tree-datasource.service.ts (hunk 2 above)
Comment (swapnil-verma-gl) [reply lib/content-services/src/lib/category/services/category-tree-datasource.service.ts:81]: Oh, I was not aware that there was backend JIRA already raised for this task. In that case, I'll close this PR, and we'll raise a new one once that task is complete. I've also linked the 2 JIRAs together. Thanks!

Comment (AleksanderSklorz) [reply lib/content-services/src/lib/category/services/category-tree-datasource.service.ts:81]: ok, thank you! :)

**Pull request closed without merging**